refnx/reduce/reduce.py

```diff
@@ -1098,6 +1131,363 @@ def splice_datasets(self, ds):
+    pm : PlatypusNexus object
+        Polarised spectra for +- spin channel
+    mp : PlatypusNexus object
+        Polarised spectra for -+ spin channel
+    mm : PlatypusNexus object
+        Polarised spectra for -- spin channel
+
+    Returns
+    -------
+    I00, I01, I10, I11 : PlatypusNexus object
+    with the efficiency-corrected spectrum
+    """
+    # The values used below are from Thomas Saerbeck's initial
+    # commissioning of the PLATYPUS polarisation system.
+    # TODO: Create a workflow that automatically fits
+    # commissioning data with an efficiency function to be used
+    # in the polarised_correction routine.
+
+    # Define polariser efficiency as function of wavelength
+    p1a = 0.993
+    p1b = 0.57
+    p1c = 0.47
+    P1 = p1a - p1b * p1c ** (wavelength).astype("float")
+
+    # Define analyser efficiency as function of wavelength
+    p2a = 0.993
+    p2b = 0.57
+    p2c = 0.51
+    P2 = p2a - p2b * p2c ** (wavelength).astype("float")
+
+    # Define flipper1 and flipper2 efficiencies
+    F1 = np.full((len(wavelength), 1), 0.003).astype("float")
+    F2 = np.full((len(wavelength), 1), 0.003).astype("float")
+
+    if I01 is None and I10 is not None:
+        # For the case when only one spin-flip channel is recorded
+        I01 = I10
+
+    if I01 is None and I10 is None:
+        # For the case when only R++ and R-- are recorded
+        F2 = np.full((len(wavelength), 1), 0.000)
+        P2 = np.full((len(wavelength), 1), 1.000)
+
+    P1 = (1 + P1) / 2
+    P2 = (1 + P2) / 2
+
+    # Define polarised efficiency matrix
+    polariser_eff_matrix = [
+        np.array(
+            [
+                [(1 - p1l), 0, p1l, 0],
+                [0, (1 - p1l), 0, p1l],
+                [p1l, 0, (1 - p1l), 0],
+                [0, p1l, 0, (1 - p1l)],
+            ]
+        )
+        for p1l in P1
+    ]
+
+    # Define analyser efficiency matrix
+    analyser_eff_matrix = [
+        np.array(
+            [
+                [(1 - p2l), p2l, 0, 0],
+                [p2l, (1 - p2l), 0, 0],
+                [0, 0, (1 - p2l), p2l],
+                [0, 0, p2l, (1 - p2l)],
+            ]
+        )
+        for p2l in P2
+    ]
+    # Define flipper 1 efficiency matrix
+    flipper1_eff_matrix = [
+        np.array(
+            [
+                [1, 0, 0, 0],
+                [0, 1, 0, 0],
+                [f1l, 0, (1 - f1l), 0],
+                [0, f1l, 0, (1 - f1l)],
+            ]
+        ).astype("float")
+        for f1l in F1
+    ]
+    # Define flipper 2 efficiency matrix
+    flipper2_eff_matrix = [
+        np.array(
+            [
+                [1, 0, 0, 0],
+                [0, 1, 0, 0],
+                [f2l, 0, (1 - f2l), 0],
+                [0, f2l, 0, (1 - f2l)],
+            ]
+        ).astype("float")
+        for f2l in F2
+    ]
+
+    # Create efficiency matrices from individual
+    # component efficiency matrices
+    efficiencies_matrix = [
```

Comment:
I think this can be vectorised. If `flipper1_eff_matrix, flipper2_eff_matrix, polariser_eff_matrix, analyser_eff_matrix` all have the same shape `(wavelength.size, 4, 4)` then I think one can just do:
 
```
flipper1_eff_matrix @ flipper2_eff_matrix @ polariser_eff_matrix @ analyser_eff_matrix
```